homeworks/07-mars-viewer/src/components/marsViewer/gallery/galleryUtils.ts

```diff
@@ -0,0 +1,13 @@
+import { Dispatch } from 'react'
+import { favouriteStatusChanged } from '../../../types/action'
+
+export const changeFavouriteStatus = (dispatch: Dispatch<any>, chosenImage: string, favourites: Set<string>) => {
+    let arr: string[] = Array.from(favourites)
+
+    if (favourites.has(chosenImage))
+        arr = arr.filter((image: string) => image !== chosenImage)
+    else
+        arr.push(chosenImage)
```

Comment:
this logic should be in reducer, since reducer is pure function it will be easier to test this part of code